### PR TITLE
add registration captcha

### PR DESCRIPTION
**Pull request by kevintsoii**

### CODE REVIEW

**Review by knhn1004 — APPROVED** [review]
LGTM

File: api/main_endpoints/util/captcha.js [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+const logger = require('../../util/logger');
+const config = require('../../config/config.json');
+
+async function verifyCaptcha(responseToken) {
+  const secretKey = config.googleApiKeys.SECRET_KEY;
+
+  try {
+    const response = await axios.post(
+      `https://www.google.com/recaptcha/api/siteverify?secret=${secretKey}&response=${responseToken}`,
+    );
```
Comment (evanugarte) [inline api/main_endpoints/util/captcha.js]: somewhat cleaner
```suggestion
    const url = new URL('https://www.google.com/recaptcha/api/siteverify');
    url.searchParams.append('secret', secretKey);
    url.searchParams.append('response', responseToken);

    const response = await axios.post(url.href);
```

Comment (kevintsoii) [reply api/main_endpoints/util/captcha.js]: done

File: api/config/config.example.json [hunk 2]
```diff
@@ -18,7 +18,8 @@
       "https://developers.google.com/oauthplayground"
     ],
     "REFRESH_TOKEN": "NOT_SET",
-    "USER": "sce.sjsu@gmail.com"
+    "USER": "sce.sjsu@gmail.com",
+    "SECRET_KEY": "NOT_SET"
```
Comment (evanugarte) [inline api/config/config.example.json]:
```suggestion
    "CAPTCHA_SECRET_KEY": "NOT_SET"
```

File: api/main_endpoints/util/captcha.js [hunk 3]
```diff
@@ -0,0 +1,23 @@
+const axios = require('axios');
+
+const logger = require('../../util/logger');
+const config = require('../../config/config.json');
+
+async function verifyCaptcha(responseToken) {
+  const secretKey = config.googleApiKeys.SECRET_KEY;
```
Comment (evanugarte) [inline api/main_endpoints/util/captcha.js]:
```suggestion
  const secretKey = config.googleApiKeys. CAPTCHA_SECRET_KEY;
```

**Review by evanugarte — APPROVED** [review]

**Review by gursheyss — APPROVED** [review]

**Pull request merged**